### PR TITLE
feat: add a flag to switch where to get the data needed for the scheduled job

### PR DESCRIPTION
**Pull request by yk634**
## Context

Change following https://github.com/screwdriver-cd/queue-service/pull/58.
The change to not use `periodicBuildConfigs` has a large impact and should be selectable by a feature flag.

## Objective

Add the setting `periodicBuildTableEnabled` to allow control over whether or not `periodicBuildConfigs` are used.

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/queue/scheduler.js [hunk 1]
```diff
@@ -221,7 +247,7 @@ async function startPeriodic(executor, config) {
     }
     logger.info(`added to delayed queue for job ${job.id}`);
 
-    if (triggerBuild && !job.archived) {
+    if (triggerBuild && (periodicBuildTableEnabled || !job.archived)) {
```
Comment (DekusDenial) [inline plugins/queue/scheduler.js:250]:
```suggestion
    if (triggerBuild && !job.archived) {
```
what it was is fine for checking archived job and is not really related to `periodicBuildTable`

**Review by DekusDenial — CHANGES_REQUESTED** [review]
other than that one comment, else LGTM

File: config/custom-environment-variables.yaml [hunk 2]
```diff
@@ -237,6 +237,8 @@ queue:
       tls: REDIS_TLS_ENABLED
     slotsRefreshTimeout: REDIS_CLUSTER_SLOTS_REFRESH_TIMEOUT
   prefix: REDIS_QUEUE_PREFIX
+    # whether or not to retrieve from redis that the data needed to start periodic builds
```
Comment (y-oksaku) [inline config/custom-environment-variables.yaml:240]: Indentation looks off.

File: plugins/queue/scheduler.js [hunk 3]
```diff
@@ -111,9 +112,15 @@ async function postBuildEvent(executor, eventConfig) {
...
-    return executor.queueBreaker.runCommand('delDelayed', executor.periodicBuildQueue, 'startDelayed', [
+    await executor.queueBreaker.runCommand('delDelayed', executor.periodicBuildQueue, 'startDelayed', [
         { jobId: config.jobId }
     ]);
+
+    if (periodicBuildTableEnabled) {
+        return executor.redisBreaker.runCommand('hdel', executor.periodicBuildTable, config.jobId);
+    }
+
+    return Promise;
```
Comment (y-oksaku) [inline plugins/queue/scheduler.js:123]:
```suggestion
    return Promise.resolve();
```

**Review by y-oksaku — APPROVED** [review]

**Review by ibu1224 — APPROVED** [review]

**Review by DekusDenial — APPROVED** [review]

**Pull request merged**